src/SimpleAzure.Storage.HybridQueue/Helpers.cs

```diff
@@ -13,4 +16,16 @@ internal static class Helpers
         type.IsPrimitive ||
         type == typeof(string) ||
         type == typeof(decimal);
+
+    /// <summary>
+    /// Has the same intention as the null suppression operator (!) but throws an exception at the use site,
+    /// rather than the use site which occurs some unknown time later.

```

Comment:
What does this mean?

"at the use site, rather than the use site .. " ??

Reply:
Sorry, words hard. Typing this on my phone so words even harder.

Consider this:

```
var x = GetFoo()!;

collection.Add(x);

// some time later
collection[0].ToString();
```

With that code, you get a null ref exception that says something in your collection is null. Great. Now what?

Change the `!` to `.AssumeNotNull()` and you now get an exception that says "GetFoo() returned a null which was not expected" and you have something to go on, to start investigating the bug.

So this moves the exception to be explicit, at the place where you are confident there shouldn't be a null, versus sometime later whenever the errant null happens to be derefenced

Reply:
Gotcha.

I prefer _not_ to add code and rely on what the language has provided (if possible) but this feels like a legit case.

Again, it also feels like a _common use case_ which would be nice that this extra verbose ceremony wasn't required but part of the status quo.

Approved.